library/scripts/jjb_compare_xml.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+set -ex
+
+# workaround for old RHELs, we need to install correct tox system-wide
+rm -rf venv
+virtualenv venv
+source venv/bin/activate
+pip install pip --upgrade
+pip install tox
+# ---
+
+JOBS_OUT_DIR=${WORKSPACE}/output/jobs
+JOBS_LOGFILE=${JOBS_OUT_DIR}/jobs-diff.log
+RESULT=''
+BLOCKLIST=blocklist
+
+# First generate output from BASE_COMMIT vars value
+git checkout "${targetBranch}"
+
+rm -rf .tox
+for ENV in ${WORKSPACE}/servers/*; do
+  tox -e "${ENV##*/}"
+  mkdir -p "./output/jobs/old"
+  mv "./output/${ENV##*/}" "./output/jobs/old/${ENV##*/}"
+done
+
+# Then use that as a reference to compare against HEAD
+git checkout "origin/pr/${pullRequestId}/merge"
+rm -rf .tox
+for ENV in ${WORKSPACE}/servers/*; do
+  tox -e "${ENV##*/}"
+  mkdir -p "./output/jobs/new"
+  mv "./output/${ENV##*/}" "./output/jobs/new/${ENV##*/}"
+done
+
+compare_xml() {
```

Comment:
the function declaration break flow of main logic, and it is hard to read this script.
It would be better to move this function in top of file or just remove it, because there is only one place where this function is called.